packages/editor/src/pages/Auth/index.tsx

```diff
@@ -141,19 +137,32 @@ class AuthPage extends React.Component<IProps, IState> {
         };
       }
 
-      if (this.state.publicKey && !this.state.isIE) {
-        const random = generateCryptoSafeRandom();
-
-        sessionStorage.setItem(
-          AUTH_PAGE_SESSION_STORAGE_KEYS.auth_state,
-          random.toString(),
-        );
+      if (this.state.publicKeyString && !this.state.isIE) {
+        // Before navigating away, kick off a process to ensure that the key is
+        // actually valid (e.g., that it wasn't accidentally cut off during copy-paste)
+        reconstructPublicKey(this.state.publicKeyString)
+          .then(_ => {
+            const random = generateCryptoSafeRandom();
```

Comment:
I realize this is existing code, but [generateCryptoSafeRandom()](https://github.com/OfficeDev/script-lab/blob/master/packages/common/src/utilities/misc.ts#L13) looks like it generates a 32-bit state, which may be a little short to defend against CSRF. I looked for formal external guidance on this, best I could find was [FHIR](http://hl7.org/fhir/smart-app-launch/#step-1-app-asks-for-authorization) which recommends 128 bits for the state parameter. This happens to be in line with internal guidance we would give around random tokens.

Reply:
@scovetta, thanks for the diligence :-).  Can I just apply that function 4 times and concatenate the result to get to 128 bits?

Reply:
Yes, you can run it four times. (There'll be a little extra overhead in the repeated calls to getRandomValues, but probably insignificant.)

I wouldn't just concatenate the numbers, as I'm pretty sure there'll be some entropy loss (e.g. [A=1, B=2, C=3, D=456] and [A=12, B=34, C=5, D=6] would both have the same output string). Instead, maybe just throw each output into a buffer at a (multiple of 32)-bit offset?